pkg/restore/restic_restore_action.go

```diff
@@ -76,6 +76,11 @@ func (a *ResticRestoreAction) Execute(input *velero.RestoreItemActionExecuteInpu
 		return nil, errors.Wrap(err, "unable to convert pod from runtime.Unstructured")
 	}
 
+	var sourcePod corev1.Pod
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(input.ItemFromBackup.UnstructuredContent(), &sourcePod); err != nil {
+		return nil, errors.Wrap(err, "unable to convert source pod from runtime.Unstructured")
+	}
+
```

Comment:
Just for reference, we could have also reverse looked up the `RestoreItemActionExecuteInput.Restore.Spec.NamespaceMapping`. But this is also fine.

Reply:
I thought about doing that but there was actually an issue with that approach. At the point when this action's `Execute` function is [called in `restoreItem`](https://github.com/vmware-tanzu/velero/blob/main/pkg/restore/restore.go#L1029-L1033), the namespace mapping hasn't yet been applied to the object being restored. That doesn't happen until [later in the function](https://github.com/vmware-tanzu/velero/blob/main/pkg/restore/restore.go#L1120-L1125). This means that, under the current implementation, both `pod` and `sourcePod` have the same namespace when this is called. Rather than assuming they would be the same, I opted to look at the item from the backup. It seemed too risky for this change to change when the namespace mapping would be applied in `restoreItem`, but checking the pod from the backup also prevents this issue occurring again if when we do change when the mapping is applied.

Reply:
I should add a comment though to explain why we can't rely on the namespace mapping from the restore spec. Thanks for reminding me about why I couldn't use that approach! :+1:

Reply:
@zubron Not sure I follow. The `NamespaceMapping` that I was referring to is a field on the restore object spec. This field is set using the value of the [`namespace-mappings`](https://github.com/vmware-tanzu/velero/blob/main/pkg/cmd/cli/restore/create.go#L112) flag that is passed at the CLI running `velero restore create` command.

Reply:
Thanks for clarifying @zubron.
Just capturing our discussion here. Changes at [line79-82](https://github.com/vmware-tanzu/velero/pull/3475/files#diff-9f110ab28d31cd60e15e9ec43ad813f22bd8b9329eaf0ebe70c0b6590592b76dR79-R87) weren't necessary and we could have used `pod.Namespace` at https://github.com/vmware-tanzu/velero/pull/3475/files#diff-9f110ab28d31cd60e15e9ec43ad813f22bd8b9329eaf0ebe70c0b6590592b76dR100 
and added unit tests to catch any future code reordering that may break. The changes at [line79-82](https://github.com/vmware-tanzu/velero/pull/3475/files#diff-9f110ab28d31cd60e15e9ec43ad813f22bd8b9329eaf0ebe70c0b6590592b76dR79-R87) is future proofing.